### PR TITLE
added support for update release name command

### PR DESCRIPTION
**Pull request by PranaySPatil**

### CODE REVIEW

File: src/Agent.Worker/Release/ReleaseCommandExtension.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    public sealed class ReleaseCommandExtension : AgentService, IWorkerCommandExtension
+    {
+        public Type ExtensionType => typeof(IWorkerCommandExtension);
+
+        public string CommandArea => "release";
+
+        public HostTypes SupportedHostTypes => HostTypes.All;
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseCommandExtension.cs]: we only should support Release & Deployment host types

File: src/Agent.Worker/Release/ReleaseServer.cs [hunk 2]
```diff
@@ -32,5 +34,15 @@ public IEnumerable<AgentArtifactDefinition> GetReleaseArtifactsFromService(int r
             var artifacts = _releaseHttpClient.GetAgentArtifactDefinitionsAsync(_projectId, releaseId, cancellationToken: cancellationToken).Result;
             return artifacts;
         }
+
+        public async Task<RMContracts.Release> UpdateReleaseName(
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseServer.cs]: UpdateRelease would require permission to manage releases

When agent runs this command, I believe it runs under the Build Service identity context. I don't think this identity has permission to update. We need to make change in RM service to grant this permission for this identity.

File: src/Agent.Worker/Release/ReleaseCommandExtension.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            Guid projectId = context.Variables.System_TeamProjectId ?? Guid.Empty;
+            ArgUtil.NotEmpty(projectId, nameof(projectId));
+            
+            string releaseId = context.Variables.Release_ReleaseId;
+            ArgUtil.NotNull(releaseId, nameof(releaseId));
+
+            if (!String.IsNullOrEmpty(data))
+            {
+                // update release name within Context.
+                context.Variables.Set("release.releaseName", data);
```
Comment (omeshp) [inline src/Agent.Worker/Release/ReleaseCommandExtension.cs]: set this after async command succeedes?

File: src/Agent.Worker/Release/ReleaseCommandExtension.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+{
+    public sealed class ReleaseCommandExtension : AgentService, IWorkerCommandExtension
+    {
+        public Type ExtensionType => typeof(IWorkerCommandExtension);
+
+        public string CommandArea => "release";
+
+        public HostTypes SupportedHostTypes => HostTypes.Release | HostTypes.Deployment;
+
+        public void ProcessCommand(IExecutionContext context, Command command)
```
Comment (omeshp) [inline src/Agent.Worker/Release/ReleaseCommandExtension.cs]: update https://github.com/Microsoft/vsts-tasks/blob/master/docs/authoring/commands.md ?

**Comment by t-moe** [comment]
Looks like this closes #1303.

However, after upgrading my agent to version 2.134.2 it doesn't work.

@GitHubSriramB 
>  We need to make change in RM service to grant this permission for this identity.

Is the change in question a server-side change? Do I need to upgrade my tfs installation? (currently on tfs 2018 update 1)

**Pull request merged**